### DISCUSSION TITLE
How are you handling huge migration folders

### DISCUSSION THREAD

**Discussion by paulwer** [prisma/prisma#24927] [Q&A] [answered]
### Question

Hi everyone,
I've asked a Feature Request a while ago for quashing migrations in favor of creating a schema dump, to reduce count of migration files and initial database creation times.
https://github.com/prisma/prisma/issues/24637

How are you handling this issue in your projects?
Are there any solutions out there already?

**Comment by ludralph** [comment]
Hi @paulwer 👋 

Thank you for creating that feature request.

At the moment, Prisma does support squashing migrations.  [Squashing migrations](https://www.prisma.io/docs/orm/prisma-migrate/workflows/squashing-migrations) can help with your use case of reducing the number of migration files and improving initial database creation times. Squashing migrations involves combining multiple migration files into a single migration, which can simplify the migration history and make the process more efficient.

 Have you considered this approach pending when the schema dump feature is available?

**Reply by paulwer** [reply]
Hi @ludralph,
the process of squashing migration is known to me. As I mentioned, other db-migration handlers like laravel already provide this feature by default.

I just wanted to know, if there are any other solutions / experiences to achieve this manualy without a prisma-command or simmilar.
f.ex. removing _prisma_migrations table / mark migrations in any way and create initial migration with dump of database.

We are currently try to find a way to reduce our migration count, because its steadily growing and will consume quite some time to execute. In a few months/year this could result in quite long times, which are not acceptable then.

Can you please explain your last question, as of I am not understanding it completly. Sorry :)

Thanks
Paul

**Reply by ludralph** [reply]
Hi @paulwer, 

Thank you for providing more context. My last question was suggesting that, while waiting for a potential schema dump feature to be implemented in Prisma, you might consider using the approach of squashing migrations to address the issue of having too many migration files in the interim.

**Reply by paulwer** [reply]
Just squashing migrations wont work if i am right, as of the deploy would fail?!

how have you done this in details? in short: how to handle prisma in this case, so it will not report an error

**Reply by paulwer** [reply]
@ludralph in short: I wanna know if someone has a step by step plan for doing such thing :)

**Reply by ludralph** [reply, marked as answer]
Hi

You may want to take a look at this [issue](https://github.com/prisma/prisma/issues/5714) for squashing migrations.

**Reply by paulwer** [reply]
Hi, I had and I will look into it again in detail.
I would also encourage to explain this within the original docs in details, as of others will encounter this problem on larger projects as well, to have a clean quick start for this.

Thanks

**Reply by ludralph** [reply]
Thanks for the feedback. I'll incorporate this into the documentation.

**Reply by paulwer** [reply]
Hi @ludralph,
thanks for the effort taken :)
Does this aproach work for all custom applied changes?
f.ex. stored procedures or triggers, we dont do this but i also have seen people using migrations for seeders

If the diff aproach does not handle these, I think there should be a section like limitations.
